utils/utils.go

```diff
@@ -448,9 +448,13 @@ func createNetworkPolicyPeer(appName string, namespace string, networkPolicy com
 
 func customizeNetworkPolicyPorts(ingress *networkingv1.NetworkPolicyIngressRule, ba common.BaseComponent) {
 	var ports []int32
-	ports = append(ports, ba.GetService().GetPort())
+	primaryTargetPort := ba.GetService().GetPort()
+	if ba.GetService().GetTargetPort() != nil {
+		primaryTargetPort = *ba.GetService().GetTargetPort()
+	}
+	ports = append(ports, primaryTargetPort)
 	for _, port := range ba.GetService().GetPorts() {
-		ports = append(ports, port.Port)
+		ports = append(ports, int32(port.TargetPort.IntValue()))
```

Comment:
NetworkPolicy supports named ports, so we can youse `port.TagetPort` as is without getting int value.
Our CRD I think can have something like this:

```
ports:
- targetPort: http
````
Above code I think would fail.  Also can you make sure `port.TargetPort` is always set? in spec its marked as optional. Might be good to wrap into nil check even if the api always sets targetPort at the moment

Reply:
Thanks, I reorganized the code to use an `intstr.IntOrString` type allowing me to just use `port.TargetPort`. Also, I added an empty string check on the target port.